### PR TITLE
Timeseries split

### LINKED ISSUE #409: Time-series split for cross validation not available in Spark ML library
Scikit learn includes a useful time-series split for cross validation. Importantly, this split does not shuffle data and ensures test sets  follow training sets in time. See here: https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.TimeSeriesSplit.html

A similar feature is not available in Spark ML. I have written an implementation here (may not be very efficient):

def time_series_split(output, n_splits):
    # creates a new column in Spark DF 'output' called 'row_num'
    w = Window().partitionBy(lit('a')).orderBy(lit('a'))
    output = output.withColumn("row_num", row_number().over(w))

    # creates three lists of indices for 'row_num'
    num_rows = output.count()
    num_rows_per_split = int(num_rows / (n_splits + 1))

    end_train_row_list = []
    start_test_row_list = []
    end_test_row_list = []
    for i in range(1, n_splits + 1):
        end_train_row_list.append(i * num_rows_per_split + 1)
        start_test_row_list.append(i * num_rows_per_split + 2)
        end_test_row_list.append(i * num_rows_per_split + 2 + num_rows_per_split)

    # creates new Spark DFs using lists of indices for 'row_num'
    train_data = {}
    test_data = {}
    for i in range(n_splits):
        train_data[f"train_data_{i}"] = output.filter(col("row_num").between(1, end_train_row_list[i]))
        test_data[f"test_data_{i}"] = output.filter(col("row_num").between(start_test_row_list[i], end_test_row_list[i]))

    return train_data, test_data

Such a feature in Spark ML would be very useful.

### PR DESCRIPTION
**Pull request by tnixon**
## Changes
Created a sub-class of the PySpark ML CrossValidator 
(see https://spark.apache.org/docs/latest/api/python/reference/api/pyspark.ml.tuning.CrossValidator.html#pyspark.ml.tuning.CrossValidator)
that replicates the timeseries split method implemented by SKLearn's TimeSeriesSplit 
(see https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.TimeSeriesSplit.html)

### Linked issues
Resolves #409 

### Functionality 

- [ ] added relevant user documentation
- [X] added a new Class method
- [ ] modified existing Class method: `...`
- [ ] added a new function
- [ ] modified existing function: `...`
- [ ] added a new test
- [ ] modified existing test: `...`
- [X] added a new example
- [ ] modified existing example: `...`
- [ ] added a new utility
- [ ] modified existing utility: `...`

### Tests
I created an example notebook where I use the new Cross-Validator to train a GBT regression model on some sample data. Notebook included as a reference example.

- [X] manually tested
- [ ] added unit tests
- [x] added integration tests
- [ ] verified on staging environment (screenshot attached)

### CODE REVIEW

**Review by R7L208 — CHANGES_REQUESTED** [review]
The changes look good. Can you add test cases for `TimeSeriesCrossValidator` @tnixon ?

**Comment by tnixon** [comment]
Yeah... I guess that would be a good idea 🙄 😄
